File: languages/julia/gladiator.jl
s = read(stdin, String)
out = ""
if occursin("\"discipline\":\"construct\"", s)
  if occursin("\"arena\":\"cheeks\"", s) out = "()" end
  if occursin("\"arena\":\"ears\"", s) out = "/\\\\/\\\\" end
  if occursin("\"arena\":\"eyes\"", s) out = "oo" end
  if occursin("\"arena\":\"geometry\"", s) out = "" end
  if occursin("\"arena\":\"head_top\"", s) out = "_" end
  if occursin("\"arena\":\"mouth\"", s) out = "^" end
  if occursin("\"arena\":\"newlines\"", s) out = "" end
  if occursin("\"arena\":\"nose\"", s) out = "." end
  if occursin("\"arena\":\"padding\"", s) out = "      " end
  if occursin("\"arena\":\"whiskers\"", s) out = "><" end
  print("{\"ok\":true,\"output\":\"$out\"}")
else
  local ok = occursin("\"candidate\":\" /\\\\_/\\\\\\n( o.o )\\n > ^ <\"", s)
  local v = ok ? "es un gato" : "no es el gato canonico"
  print("{\"ok\":", ok, ",\"output\":\"$v\"}")
end

File: languages/ruby/gladiator.rb
s = STDIN.read
out = ""
if s.include?("\"discipline\":\"construct\"")
  out = "()" if s.include?("\"arena\":\"cheeks\"")
  out = "/\\\\/\\\\" if s.include?("\"arena\":\"ears\"")
  out = "oo" if s.include?("\"arena\":\"eyes\"")
  out = "" if s.include?("\"arena\":\"geometry\"")
  out = "_" if s.include?("\"arena\":\"head_top\"")
  out = "^" if s.include?("\"arena\":\"mouth\"")
  out = "" if s.include?("\"arena\":\"newlines\"")
  out = "." if s.include?("\"arena\":\"nose\"")
  out = "      " if s.include?("\"arena\":\"padding\"")
  out = "><" if s.include?("\"arena\":\"whiskers\"")
  print '{"ok":true,"output":"' + out + '"}'
else
  ok = s.include?("\"candidate\":\" /\\\\_/\\\\\\n( o.o )\\n > ^ <\"")
  print '{"ok":' + (ok ? 'true' : 'false') + ',"output":"' + (ok ? "es un gato" : "no es el gato canonico") + '"}'
end
